packages/jest-matchers/src/matchers.js

```diff
@@ -312,6 +311,33 @@ const matchers: MatchersObject = {
     return {message, pass};
   },
 
+  toContainEqual(collection: Array<any>, value: any) {
+    const collectionType = getType(collection);
+    if (!Array.isArray(collection)) {
+      throw new Error(
+        `.toContainEqual() only works with arrays.\n` +
+        printWithType('Received', collection, printReceived),
+      );
+    }
+
+    const pass = collection.findIndex(elem => {
+      return equals(elem, value, [iterableEquality]);
+    }) != -1;
+    const message = pass
+      ? () => matcherHint('.not.toContainEqual', collectionType, 'value') + '\n\n' +
+        `Expected ${collectionType}:\n` +
+        `  ${printReceived(collection)}\n` +
+        `Not to contain value with deep checking:\n` +
```

Comment:
Instead of `with deep checking` here (and below) I would suggest "Not to contain an equal value to:` and below `To contain an equal value to:`.

What do you think?